File: 006.pl
#!usr/bin/perl

use 5.010;
use strict;
use bigint;

my $sum_of_squares = 0;
my $squared_sum = 0;

for my $num (1..100){
	$squared_sum += $num;
	$sum_of_squares += ($num * $num);
}

$squared_sum = ($squared_sum * $squared_sum);

print "$sum_of_squares\n";
print "$squared_sum\n";

my $difference = ($sum_of_squares - $squared_sum);

say "$difference";

File: 006.rs
fn squared_sum(n: i32) -> i32 {
    let sum = (n * (n+1)) / 2;
    sum * sum
}

fn sum_of_squares(n: i32) -> i32 {
    let mut acc = 0;
    for i in 1..(n+1) {
        acc += i * i;
    }
    acc
}

fn main() {
    println!("{}", squared_sum(100) - sum_of_squares(100)); 
}
